gen/exhibitor_tls_bootstrap.py

```diff
@@ -0,0 +1,121 @@
+                bootstrap node.
+            master_discovery must be static
+         DC/OS variant must be enterprise
+    """
+    checks = [
+        (lambda: config.get('exhibitor_tls_enabled', False) == 'true',
+         'Exhibitor security is disabled'),
+        (lambda: config['master_discovery'] == 'static',
+         'Only static master discovery is supported'),
+        (lambda: config['dcos_variant'] == 'enterprise',
+         'Exhibitor security is an enterprise feature'),
+        (lambda: urlparse(config['bootstrap_url']).scheme != 'file',
+         'Blackbox exhibitor security is only supported when using a remote'
+         ' bootstrap node'),
+    ]
+
+    reasons = []
+    for check in checks:
+        if not check[0]():
+            reasons.append(check[1])
+
+    return reasons
+
+
+def find_package(packages_json: str) -> str:
+    packages = json.loads(packages_json)
+    for package in packages:
+        if package.startswith(PACKAGE_NAME):
+            return package
+    raise Exception('dcoscertstrap package is not present')
+
+
+def _extract_package(package_path: str):
+    """ Extracts the dcoscertstrap package from the local pkgpanda
+    repository """
+    os.makedirs(BINARY_PATH, exist_ok=True)
+    with tempfile.TemporaryDirectory() as td:
+        subprocess.run(['tar', '-xJf', package_path, '-C', td])
+
+        shutil.move(
+            pathlib.Path(td) / 'bin' / PACKAGE_NAME,
+            pathlib.Path(BINARY_PATH) / PACKAGE_NAME)
+
+
+def _init_ca(alt_names: List[str]):
+    """ Initializes the CA (generates a private key and self signed
+    certificate CA extensions) if the resulting files do not already exist """
+    if not all(map(os.path.exists, [
+            pathlib.Path(CA_PATH) / 'root-cert.pem',
+            pathlib.Path(CA_PATH) / 'root-key.pem'])):
+        os.makedirs(CA_PATH, mode=0o0700, exist_ok=True)
+        cmd_path = pathlib.Path(BINARY_PATH) / PACKAGE_NAME
+        subprocess.run([
+            str(cmd_path), '-d', CA_PATH, 'init-ca', '--sans', ','.join(alt_names)
+        ])
+    else:
+        print('[{}] CA files already exist'.format(__name__))
+
+
+def _read_certificate() -> str:
+    with open(pathlib.Path(CA_PATH) / 'root-cert.pem') as fp:
+        return fp.read()
+
+
+def _get_ca_alt_name(config: Dict[str, Any]) -> str:
+    """ Gets the bootstrap url hostname. Used to populate the CA SAN
+    extension """
+    url = config['exhibitor_bootstrap_ca_url'] or config['bootstrap_url']
+    return urlparse(url).hostname or ""
+
+
+def initialize_exhibitor_ca(final_arguments: Dict[str, Any]):
+    if final_arguments['platform'] != 'onprem':
+        return
+
+    reasons = _check(final_arguments)
+    if reasons:
+        print('[{}] not bootstrapping exhibitor CA: {}'.format(
+            __name__, '\n'.join(reasons)))
+        final_arguments['exhibitor_ca_certificate'] = ""
+        final_arguments['exhibitor_ca_certificate_path'] = "/dev/null"
```

Comment:
Not sure whether setting invalid `arguments` makes sense in this case.

Reply:
These are actually valid arguments.